app/assets/javascripts/discourse/app/lib/hashtag-types/category.js

```diff
@@ -12,29 +12,20 @@ export default class CategoryHashtagType extends HashtagTypeBase {
     return this.site.categories || [];
   }
 
-  generateColorCssClasses(category) {
-    const generatedCssClasses = [];
-    const backgroundGradient = [`var(--category-${category.id}-color) 50%`];
-    if (category.parentCategory) {
-      backgroundGradient.push(
-        `var(--category-${category.parentCategory.id}-color) 50%`
-      );
-    } else {
-      backgroundGradient.push(`var(--category-${category.id}-color) 50%`);
-    }
-
-    generatedCssClasses.push(`.hashtag-color--category-${category.id} {
-  background: linear-gradient(-90deg, ${backgroundGradient.join(", ")});
-}`);
-
-    return generatedCssClasses;
+  generateColorCssClasses() {
```

Comment:
We can use just the `.hashtag-category-badge` CSS class and rely on CSS variables for customizing for each category, similar to how category badges already work. No need to generate more classes.